Kernel/Process.cpp

```diff
@@ -232,12 +232,14 @@ Region* Process::allocate_region_with_vmobject(VirtualAddress vaddr, size_t size
 
 bool Process::deallocate_region(Region& region)
 {
+    OwnPtr<Region> region_protector;
     ScopedSpinLock lock(m_lock);
```

Comment:
The commit message here seems inaccurate. This makes sure we drop the region with `Process::m_lock` unlocked, but has nothing to do with CPU interrupts. Or am I missing some bigger picture detail here?

Reply:
Ah; this was an `InterruptDisabler` and not a spinlock when I wrote the code (before the SMP PR got merged). But it looks like a spinlock still disables interrupts:

https://github.com/SerenityOS/serenity/blob/e7393bfb7b4cc8a256f94bf7f8333bb2d17ca4e0/Kernel/SpinLock.h#L122

So I kept this. I haven't tested if it now works without it, perhaps I'm wrong about spinlock+interrupts and it does.

Reply:
Ah, no you are right. I'm silly :)